src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -1060,19 +1057,23 @@ UCS_CLASS_INIT_FUNC(uct_tcp_sockcm_ep_t, const uct_ep_params_t *params)
 ucs_status_t uct_tcp_sockcm_ep_create(const uct_ep_params_t *params, uct_ep_h *ep_p)
 {
     uct_tcp_sockcm_ep_t *tcp_ep;
+    ucs_async_context_t *async;
     ucs_status_t status;
 
     if (params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR) {
         /* create a new endpoint for the client side */
         return UCS_CLASS_NEW(uct_tcp_sockcm_ep_t, ep_p, params);
     } else if (params->field_mask & UCT_EP_PARAM_FIELD_CONN_REQUEST) {
         tcp_ep = (uct_tcp_sockcm_ep_t*)params->conn_request;
+        async  = uct_tcp_sockcm_ep_get_cm(tcp_ep)->super.iface.worker->async;
```

Comment:
is this ok to do if inside the next function we might switch the cm for the tcp_ep?

Reply:
the case which you are describing did not work before too, the test/app have to protect the code related to both CMs like here
https://github.com/openucx/ucx/blob/5f06ebefaf5fb8d787f1d85d7643416b9689c246/test/gtest/uct/ib/test_sockaddr.cc#L244

Reply:
is it ok now though? if you are taking the lock with one cm and inside the function change the cm

Reply:
discussed offline - using "previous" async for lock and unlock.